include/albatross/src/cereal/variant.hpp

```diff
@@ -63,6 +63,10 @@ inline void save(Archive &archive, const variant<VariantTypes...> &f,
                  const std::uint32_t) {
 
   archive(cereal::make_nvp("which", f.which()));
+  f.match([&](const auto &x) {
+    std::string which_typeid = typeid(x).name();
```

Comment:
This will give an implementation defined name for the type. It's not guaranteed to be the same across compilers, or ever versions of the same compiler.  How critical is it that these names are always the same?

Reply:
I also think this tends to give the mangled name, in case that's a problem.

Reply:
Yeah that's totally fine.  I mostly added this so that when you look at serialized output you can tell which of the variants was used (since the `which` element is just an integer).

Reply:
Makes sense. I guess the `which` field is just has fragile since the order of the template parameters.